test/api_save_product_test.dart

```diff
@@ -505,13 +510,150 @@ Like that:
       expect(product.nutriments, isNull);
     });
 
-    test('Nutriments', () async {
+    test('With nutriments', () async {
       Product product = Product(
           nutriments: Nutriments.empty()
             ..setValue(Nutrient.salt, PerSize.oneHundredGrams, 1.0));
 
       expect(product.noNutritionData, isFalse);
       expect(product.nutriments, isNotNull);
     });
+
+    test('Nutriments modifiers (server call)', () async {
+      Product product = Product(
+        barcode: '1234567890123',
```

Comment:
The problem is that you're always setting the same values for the same product. So if you're lucky it may work the first time, and if you're not lucky for some reason it doesn't work anymore (e.g. server somehow changed) and you won't be able to see that.
You may rather start with a `getProductV3`, and change each modifier.